pkg/activator/handler/enforce_length_handler_test.go

```diff
@@ -13,51 +13,104 @@ limitations under the License.
 package handler
 
 import (
-	"bytes"
+	"io"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
+	"strings"
 	"testing"
 )
 
 func TestEnforceMaxContentLengthHandler(t *testing.T) {
 	payload := "SAMPLE PAYLOAD"
 
+	// httptest.NewRequest will set ContentLength for strings.NewReader
+	fixed := func(p string) io.Reader { return strings.NewReader(p) }
+	stream := func(p string) io.Reader { return ioutil.NopCloser(strings.NewReader(p)) }
+
 	examples := []struct {
 		label     string
 		maxUpload int
+		request   io.Reader
+		response  string
 		status    int
 	}{
 		{
-			label:     "under",
+			label:     "under with ContentLength",
 			maxUpload: len(payload) + 1,
+			request:   fixed(payload),
 			status:    http.StatusOK,
+			response:  payload,
 		},
 		{
-			label:     "equal",
+			label:     "equal with ContentLength",
 			maxUpload: len(payload),
+			request:   fixed(payload),
 			status:    http.StatusOK,
+			response:  payload,
 		},
 		{
-			label:     "over",
+			label:     "over with ContentLength",
 			maxUpload: len(payload) - 1,
+			request:   fixed(payload),
 			status:    http.StatusRequestEntityTooLarge,
+			response:  "",
+		},
+		{
+			label:     "under without ContentLength",
+			maxUpload: len(payload) + 1,
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload,
+		},
+		{
+			label:     "equal without ContentLength",
+			maxUpload: len(payload),
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload,
+		},
+		{
+			label:     "over without ContentLength",
+			maxUpload: len(payload) - 1,
+			request:   stream(payload),
+			status:    http.StatusOK,
+			response:  payload[0 : len(payload)-1],
 		},
 	}
 
 	for _, e := range examples {
 		t.Run(e.label, func(t *testing.T) {
+			// Return 200 and request body
 			baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				w.WriteHeader(http.StatusOK)
+
+				if r.Body != nil {
+					body, err := ioutil.ReadAll(r.Body)
+					if err != nil {
+						t.Fatalf("Error request reading body")
+					}
+					w.Write(body)
+				}
 			})
 			handler := EnforceMaxContentLengthHandler{NextHandler: baseHandler, MaxContentLengthBytes: int64(e.maxUpload)}
 
 			resp := httptest.NewRecorder()
-			req := httptest.NewRequest("POST", "http://example.com", bytes.NewBufferString(payload))
+			req := httptest.NewRequest("POST", "http://example.com", e.request)
 
 			handler.ServeHTTP(resp, req)
 
+			gotBody, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("Error request reading body")
+			}
+
+			if e.response != string(gotBody) {
```

Comment:
```
if  got, want := e.response, string(gotBody); got != want {
  t.Errorf("Body = %q, want = %q", got, want)
```
Same for code?

Reply:
Thanks for the feedback, but going to respectfully disagree. I'd like to avoid code golfing in tests, where legibility and maintainability are a priority.

Reply:
I don't see golfing here. `if got, want :=...; got != want` is a standard Go testing idiom; FWIW, what I propose is longer, so the opposite of golfing. But up to you.
For the error message, I'd still prefer the change. If test fails, we know unexpected thing happened.